### ISSUE TITLE
Audit requirements.txt and requirements_dev.txt

### ISSUE THREAD

**Issue by crcrpar** [NVIDIA/apex#1519]
## requirements.txt
- pytest is not a requirement for installation: https://github.com/NVIDIA/apex/blob/ee42e2d64f25b884823b8aa76b94c4b655ec7ea1/requirements.txt#L5

## requirements_dev.txt
- pytest: optional 
- expecttest
- hypothesis

The bottom two derive from the depndency on `torch.testing` things.

cc @eqy @Aidyn-A @ptrblck

**Comment by Aidyn-A** [comment]
I agree with the suggestion to add `pytest`, `expecttest` and `hypothesis`. Although the latter two are already required by the upstream pytorch I think it is still a good to have them just in case if they are not present for some reason. 
 @eqy, @ptrblck what do you think?
